### ISSUE TITLE
Is it possible to search typing directly in the main control without showing the dropdown list?

### ISSUE THREAD

**Issue by toni-angullo-bbt** [AhmedLSayed9/dropdown_button2#221]
We're trying to locate an item starting writing in the main control, **without showing the dropdown list**. I mean, when the dropdownbutton gets the focus, the user starts writing 'Calif'... and the items in the list that coincide with 'Colu' will be selected ('Columbia', 'Columbus' for instance). 

It would be useful, especially for desktop and web apps.

Example: in this dropdown

<img width="281" alt="image" src="https://github.com/AhmedLSayed9/dropdown_button2/assets/151815967/67306f4e-4544-489c-9729-cd1727e18e86">

with this item list:

<img width="272" alt="image" src="https://github.com/AhmedLSayed9/dropdown_button2/assets/151815967/f29866ed-d828-4ecf-b116-78dcad4385ae">

If the user starts writing 'Colu'...  (NOTE: not necessarily need to be showing the list)...

<img width="270" alt="image" src="https://github.com/AhmedLSayed9/dropdown_button2/assets/151815967/898e740f-97c9-413a-83ab-f862fb785d66">

... the list will be filtered by the search term. If there's only one item that coincides with the search term, it would be automatically selected.

Thank you!

**Comment by AhmedLSayed9** [comment]
I'm not sure I understand the UX here.
Do you have a real example or this is a special UX?

**Comment by toni-angullo-bbt** [comment]
> I'm not sure I understand the UX here. Do you have a real example or this is a special UX?

I'm talking about a "standard" behavior in which the user starts typing right after the control gets the focus, without having to drop down the list. As they're typing, the selected item is changing to reflect the first item in the list that coincides with the text that they've written so far. So they wouldn't need to expand the list of items (only by typing the correct word, they select the proper item).

It's like an **inline autocomplete**. That would be the word, I believe.

You can see an example in this link:

[https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-autocomplete-both/#ex_label](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-autocomplete-both/#ex_label)

**Comment by AhmedLSayed9** [comment]
Both features editable dropdown field and inline autocomplete are not supported yet.

**Comment by toni-angullo-bbt** [comment]
> Both features editable dropdown field and inline autocomplete are not supported yet.

Ok, I see... it's a pity, we'd have loved it if we'd been able to use it. Thank you so much, @AhmedLSayed9.

Do you think it would be interesting to implement it?

**Comment by AhmedLSayed9** [comment]
> Do you think it would be interesting to implement it?

Yes, but it'll take some time. Feel free to contribute if you want :)
